doc/arch/015-online-patient-lookup.md

```diff
@@ -0,0 +1,121 @@
+# Online Patient Lookup
+
+## Status
+
+Proposed on 2020-04-12
+
+## Context
+
+Want want to ensure that a patient’s medical records within the state
+are always available to the nurse for treatment if they are connected
+to the internet. See related
+[PRD](https://docs.google.com/document/d/1q6cppByQULfh3_mMXC4BJpiNN9Uc_awA6rreeEtUBaM/edit#)
+for more details on the feature and related specifications.
+
+The lookup will be a new API on the server that the mobile app will
+call. The following aspects of the problem are addressed in this ADR:
+
+- API contract
+- Data retention
+- Access restrictions
+- Audit logging
+- Rate limiting
+
+## Decision
+### Data retention
+There are two broad kinds of data retention on the app at a facility:
+1. Permanent: patient is within sync criteria: the same block, has an
+   appointment, or is assigned to the facility
+2. Temporary: patient is not within the sync criteria
+
+In order to have a better control of the retention, we will send this
+information from the server. This will be a static number to begin
+with, but can later vary depending on the state, country or other
+factors.
+
+We will implement temporary retention in the app with this
+feature. With every patient retrieved via the lookup API, we will:
+
+- ensure the data is synced to the server
+- delete the data after the retention time period has passed
+
+< @mobile-devs: please add details on possible approaches here >
+
+Note that the retention time period is not strict. i.e,
+it is okay for the record to stay on the phone for 2h if the retention
+period is 1h. We can perform deletions at a convenient time as long as
+the deletion happens deterministically.
+
+_Alternatively_: we can choose to hard-code the retention period on
+the app.
+
```

Comment:
@msasikanth @vinaysshenoy need your help to hash this section out! 🙇

Reply:
To begin with, I think we can go with a hard-coded (or remote config option) retention period on the app. That way we can remove the patient once the retention period is finished during our regular sync. A similar logic should apply for the retention period sent from the server, we remove the user from the local DB once the retention period is finished during our regular sync interval. I think that interval is deterministic, WDYT? 

Also, we won't remove the patient if they click on the sync indicator to avoid any confusion for them, since patients will be removed from the recent patient's list when they manually sync.

Reply:
> or remote config option

If it's not very different in terms of implementation, I'd prefer keeping this config on the API for reasons mentioned in the ADR.

> once the retention period is finished during our regular sync interval. I think that interval is deterministic, WDYT?

Yup, this works quite well. What is the current sync interval?

> since patients will be removed from the recent patient's list when they manually sync.

As long as we continue to keep the record until the retention period is completed, we should be good. I feel it might be simpler to just apply this rule of `max(sync-time, retention-period-expires-time)` irrespective of how the sync is initiated. But, your call ✌️ .

Reply:
> What is the current sync interval?

Currently it's 15 minutes

Reply:
I wouldn't worry too much about the recent patients list. Unless you're in the last 3 patients that a nurse saw, the nurse isn't going to notice a disappearing patient.

If we're really worried about folks noticing it, or relying on the recent patient list to find older patients, then we should adjust our retention period accordingly - eg. make it 1 day instead of 1 hour.

Reply:
@msasikanth I've tried to write out what the implementation details should look like. Please have a look, and suggest changes.

Reply:
Spoke to @msasikanth over a call, we will write this section out later.